### PR TITLE
Drupal 9 compatibility

### PR DESCRIPTION
**Pull request by tomspiers**
Hello!

Looking to use this module on a Drupal9 project.

Tested on fresh D8 and D9 installs and all seems to be working.

Hope you're all well at Capgemini :)

Tom Spiers

### CODE REVIEW

File: spalp.module [hunk 1]
```diff
@@ -105,7 +105,9 @@ function spalp_get_app_ids(FieldStorageConfig $definition, ContentEntityInterfac
  * Implements hook_modules_installed().
  */
 function spalp_modules_installed($modules) {
-  $module_list = system_rebuild_module_data();
+  /** @var \Drupal\Core\Extension\ModuleExtensionList $module_list_service */
+  $module_list_service = \Drupal::service("extension.list.module");
```
Comment (malcomio) [inline spalp.module]: should be consistent with using single quotes unless double quotes are needed

File: spalp_example/src/EventSubscriber/SpalpExampleConfigLocationAlterSubscriber.php [hunk 2]
```diff
@@ -34,8 +34,8 @@ public function doConfigLocationAlter(SpalpConfigLocationAlterEvent $event) {
       // TODO: proper dependency injection example.
       $module_path = drupal_get_path('module', 'spalp_example');
       $locations = [
-        'config' => $module_path . '/spalp_example.config.json',
-        'schema' => $module_path . '/spalp_example.config.schema.json',
+        'config' => $module_path . '/config/spalp/spalp_example.config.json',
```
Comment (malcomio) [inline spalp_example/src/EventSubscriber/SpalpExampleConfigLocationAlterSubscriber.php:37]: thanks - I should have sorted this out at the time of making the location change

File: spalp.module (hunk 1 above)
Comment (tomspiers) [reply spalp.module]: Done :)

**Comment by tomspiers** [comment]
@malcomio - good comment raised here https://www.drupal.org/project/spalp/issues/3215426

Would we still want to support 8.7.7 and below? If so we still need 'core: 8.x'

File: spalp_example/src/EventSubscriber/SpalpExampleConfigLocationAlterSubscriber.php (hunk 2 above)
Comment (malcomio) [reply spalp_example/src/EventSubscriber/SpalpExampleConfigLocationAlterSubscriber.php:37]: odd, this is already correct:
https://github.com/Capgemini/spalp/blob/8.x-1.x/spalp_example/src/EventSubscriber/SpalpExampleConfigLocationAlterSubscriber.php

You must have an outdated version of the code

**Pull request merged**